$NetBSD$

Do not install libperconaserverclient with the server.
Generate versioned lib symlinks on FreeBSD to supress PLIST divergence.

--- libmysql/CMakeLists.txt.orig	2017-12-28 09:28:11.000000000 +0000
+++ libmysql/CMakeLists.txt
@@ -230,12 +230,18 @@ IF(WIN32)
   LIST(APPEND LIBS auth_win_client)
 ENDIF()
 
+IF(NOT WITHOUT_SERVER)
 # LDAP authentication SASL client plugin
 MESSAGE(STATUS "Creating LDAP authentication SASL client library.")
 ADD_SUBDIRECTORY(authentication_ldap)
+ENDIF()
   
 # Merge several convenience libraries into one big perconaserverclient
+IF(WITHOUT_SERVER)
 MERGE_LIBRARIES(perconaserverclient STATIC ${LIBS} COMPONENT Development)
+ELSE()
+MERGE_LIBRARIES(perconaserverclient STATIC NOINSTALL ${LIBS} COMPONENT Development)
+ENDIF()
 
 # Visual Studio users need debug  static library for debug projects
 IF(MSVC)
@@ -275,12 +281,18 @@ ENDIF()
 IF(NOT DISABLE_SHARED)
   # Merge several convenience libraries into one big perconaserverclient
   # and link them together into shared library.
+  IF(WITHOUT_SERVER)
   MERGE_LIBRARIES(libmysql SHARED ${LIBS}
     EXPORTS ${CLIENT_API_FUNCTIONS} ${CLIENT_API_FUNCTIONS_UNDOCUMENTED}
     COMPONENT SharedLibraries)
+  ELSE()
+  MERGE_LIBRARIES(libmysql SHARED NOINSTALL ${LIBS}
+    EXPORTS ${CLIENT_API_FUNCTIONS} ${CLIENT_API_FUNCTIONS_UNDOCUMENTED}
+    COMPONENT SharedLibraries)
+  ENDIF()
   IF(UNIX)
     # libtool compatability
-    IF(CMAKE_SYSTEM_NAME MATCHES "FreeBSD" OR APPLE)
+    IF(CMAKE_SYSTEM_NAME MATCHES APPLE)
       SET(OS_SHARED_LIB_VERSION "${SHARED_LIB_MAJOR_VERSION}")
     ELSE()
       SET(OS_SHARED_LIB_VERSION
